src/chain/store/chain_store.rs

```diff
@@ -155,12 +155,22 @@ where
     pub fn put_tipset(&self, ts: &Tipset) -> Result<(), Error> {
         persist_objects(self.blockstore(), ts.block_headers().iter())?;
 
+        self.put_tipset_key(ts.key())?;
+
         // Expand tipset to include other compatible blocks at the epoch.
         let expanded = self.expand_tipset(ts.min_ticket_block().clone())?;
         self.update_heaviest(Arc::new(expanded))?;
         Ok(())
     }
 
+    /// Writes the `TipsetKey` to the blockstore for `EthAPI` queries
+    pub fn put_tipset_key(&self, tsk: &TipsetKey) -> Result<(), Error> {
+        self.blockstore()
+            .put_keyed(&tsk.cid()?, &fvm_ipld_encoding::to_vec(&tsk)?)?;
```

Comment:
I have checked how this is done in Lotus and it seems good. They are using `chainLocalBlockstore` for cache, but that is that same `blockstore`, unless it's configured with `LOTUS_ENABLE_CHAINSTORE_FALLBACK` flag.